### PR TITLE
chore(alerts|kube-monitoring): managed prometheus-operator and CRDs

### PR DESCRIPTION
**Pull request by richardtief**
Installation of the CustomResourceDefinitions and the prometheus operator is a unique task for a cluster. We have two plugins that depend on it and therefore need to be able to bring the basis for their installation themselves. Therefore we decided to use the functionality of the kube-prometheus-stack to manage CRDs and turn it on by default. This makes chart testing considerably easier.

# Submit a pull request

Thank you for submitting a pull request!  
To speed up the review process, please ensure that everything below
is true:

1. This is not a duplicate of an existing Plugin.
2. No existing features have been broken without good reason.
3. The Documentation has been updated to reflect your changes.
4. Tests have been added or updated to reflect your changes.
5. All tests pass.

---

Replace any ":question:" below with information about your pull request.

## Pull Request Details

Provide details about your pull request and what it adds, fixes, or changes.

:question:

## Breaking Changes

Describe what features are broken by this pull request and why, if any.

:question:

## Issues Fixed

Enter the issue numbers resolved by this pull request below, if any.

1. :question:

## Other Relevant Information

Provide any other important details below.

:question:

### CODE REVIEW

File: alerts/plugindefinition.yaml [hunk 1]
```diff
@@ -38,6 +38,14 @@ spec:
       required: false
       type: list
 
+    - name: alerts.prometheusOperator.enabled
+      description: Deploy Prometheus Operator if kube-monitoring has not already installed it once.
+      required: false
+      type: bool
+    - name: alerts.crds.enabled
+      description: Install Prometheus Operator CRDs if kube-monitoring has not already installed them.
```
Comment (ibakshay) [inline alerts/plugindefinition.yaml:46]: Do we need to set a default value for `alerts.crds.enabled`?

Comment (richardtief) [reply alerts/plugindefinition.yaml:46]: Yes. We enable it by default in the chart values, but disable it by default for the plugin as we intentionally want kube-monitoring to manage the operator and the CRDs. These options only highlight the possibility for the plugin to act independently

File: alerts/plugindefinition.yaml [hunk 2]
```diff
@@ -38,6 +38,14 @@ spec:
...
+      required: false
+      type: bool
```
Comment (ibakshay) [inline alerts/plugindefinition.yaml:48]: >Yes. We enable it by default in the chart values, but disable it by default for the plugin as we intentionally want >kube-monitoring to manage the operator and the CRDs. These options only highlight the possibility for the >plugin to act independently

@richardtief, Thank you for the explanation. Do we need to set this default value to `false` explicitly like below then? 
Or, Am I missing something? 
```suggestion
      required: false
      default: false
      type: bool
```

File: alerts/plugindefinition.yaml [hunk 3]
```diff
@@ -38,6 +38,15 @@ spec:
       required: false
       type: list
 
+    - name: alerts.prometheusOperator.enabled
+      description: Deploy Prometheus Operator if kube-monitoring has not already installed it once.
+      required: false
+      type: bool
```
Comment (ibakshay) [inline alerts/plugindefinition.yaml:44]:
```suggestion
      required: false
      default: false
      type: bool
```

**Review by ibakshay — APPROVED** [review]
LGTM! 🚀

**Review by viennaa — COMMENTED** [review]
What about updating the documentation of kube-monitoring and alerts to reflect this change? Right now kube-monitoring is stating to deploy the operator and having alerts first is now a hard dependency, no?

**Comment by richardtief** [comment]
No, the behaviour should remain the same. We are just enabling the alerts plugin to operate independently of the kube-monitoring plugin. The documentation thus remains valid - but we should still do a review of the documentation, fair point.

**Pull request merged**